ports/servoshell/desktop/headed_window.rs

```diff
@@ -415,8 +415,6 @@ impl WindowPortsMethods for Window {
                         (dx as f64, (dy * LINE_HEIGHT) as f64, WheelMode::DeltaLine)
                     },
                     MouseScrollDelta::PixelDelta(position) => {
-                        let position: LogicalPosition<f64> =
-                            position.to_logical(self.device_hidpi_factor().get() as f64);
```

Comment:
Okay. After a bit of investigation, I think I know what is going on here. The value expected by Servo is a value in logical pixels (`LayoutVector2D`). The value we are getting from `winit` is in physical pixels, so we should do a conversion. The issue here is that the factor returned by `device_hidpi_factor()` is the scale to convert from logical pixels to device pixels and `fn to_logical` wants the opposite (because it is dividing). I believe the correct thing to do here is to use the inverse of the factor. I've confirmed that this doesn't speed up scrolling too much and it looks a lot more natural:

```rust
                            position.to_logical(self.device_hidpi_factor().inverse().get() as f64);
```

In addition, this probably won't change behavior on Linux users with a mouse, because that goes through the `LineDelta` code path above.

Reply:
Ah, so scrolling would actually have been 4x too slow on my device. No wonder it felt slow! I've updated this PR to use the inverse scale factor :)

Reply:
> In addition, this probably won't change behavior on Linux users with a mouse, because that goes through the LineDelta code path above.

Yeah, I think fancy mice with "high precision" scrolling might also use the PixelDelta, but most mice are using the LineDelta.

Reply:
I suspect there is also an issue with the line delta, because when modifying the screen scaling, it scrolls different amounts even though this theoretically should be the same number of logical pixels. I think more investigation is necessary here.

Reply:
> I suspect there is also an issue with the line delta, because when modifying the screen scaling, it scrolls different amounts even though this theoretically should be the same number of logical pixels. I think more investigation is necessary here.

Seems plausible. I don't think servoshell got much attention until recently. Unfortunately I don't have device a mouse available to test that with atm, but perhaps we could create an issue.